flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/savepoints/SavepointDisposalRequest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages.job.savepoints;
+
+import org.apache.flink.runtime.rest.messages.RequestBody;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Request body for a savepoint disposal call.
+ */
+public class SavepointDisposalRequest implements RequestBody {
+
+	private static final String FIELD_NAME_SAVEPOINT_PATH = "savepoint-path";
+
+	@JsonProperty(FIELD_NAME_SAVEPOINT_PATH)
+	private final String savepointPath;
+
+	@JsonCreator
+	public SavepointDisposalRequest(
+		@JsonProperty(FIELD_NAME_SAVEPOINT_PATH) @Nonnull String savepointPath) {
+		this.savepointPath = savepointPath;
+	}
+
+	public String getSavepointPath() {
```

Comment:
I would generally recommend annotating this method with `@JsonIgnore`. Getters that are not annotaed can interfere with the `@JsonProperty` annotation.

Reply:
Good catch, will addi t.